src/game/stage/background/zone_3.c

```diff
@@ -144,13 +149,15 @@ void StageBgUpdateZone3Acts12(s32 a, s32 b)
 
     if (IS_SINGLE_PLAYER) {
         if ((gPlayer.moveState & MOVESTATE_8000000) && (gUnknown_030054F4 >= 7)) {
-            if (gUnknown_03000408 == 0) {
-                gUnknown_03000408 = a;
+            Unk3000408 *unk = IWRAM_PTR(NULL);
```

Comment:
Nice, if this matches it seems more legit. IDK

Reply:
Well it doesn't rn, but I'm not planning to spend more time on this specific one for now! :D